activerecord/lib/active_record/connection_adapters/schema_cache.rb

```diff
@@ -21,22 +22,25 @@ def initialize_dup(other)
         @columns_hash = @columns_hash.dup
         @primary_keys = @primary_keys.dup
         @data_sources = @data_sources.dup
+        @indexes      = @indexes.dup
       end
 
       def encode_with(coder)
-        coder["columns"] = @columns
+        coder["columns"]      = @columns
         coder["columns_hash"] = @columns_hash
         coder["primary_keys"] = @primary_keys
         coder["data_sources"] = @data_sources
-        coder["version"] = connection.migration_context.current_version
+        coder["indexes"]      = @indexes
+        coder["version"]      = connection.migration_context.current_version
       end
 
       def init_with(coder)
-        @columns = coder["columns"]
+        @columns      = coder["columns"]
         @columns_hash = coder["columns_hash"]
         @primary_keys = coder["primary_keys"]
         @data_sources = coder["data_sources"]
-        @version = coder["version"]
+        @indexes      = coder["indexes"]
```

Comment:
`@indexes      = coder["indexes"] || {}`

to make this work with schema caches dumped in versions that didn't know about indexes.